crates/matrix-sdk/src/authentication/oauth/auth_code_builder.rs

```diff
@@ -73,6 +73,17 @@ impl OAuthAuthCodeUrlBuilder {
         self
     }
 
+    /// Set a generic login hint to help an identity provider pre-fill the login
+    /// form.
+    ///
+    /// Note: This is not the same as the [`Self::user_id_hint()`] method, which
+    /// is specifically designed to a) take a `UserId` and no other type of
+    /// hint and b) be used directly by MAS and not the identity provider.
```

Comment:
Hmm, this overwrites whatever we set with `user_id_hint()`, we should mention this.

Out of interest, what will you set as the hint?

Reply:
Good point, I've added the same `The following methods are mutually exclusive` line that we use in the client builder.

It will likely be an email address but it depends on the upstream provider, we would pass through whatever is given to us without modifying it. The use case is an organisation creating provisioning links like `https://app.example.com/?server_name=example.org&login_hint=alice` where their upstream provider knows how to handle `alice` as a hint.

There are docs here: https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest

Reply:
Ah makes sense. Can we perhaps more explicitly mention this example use-case and link to these docs?